pkg/test/framework.go

```diff
@@ -101,6 +103,8 @@ type addToSchemeFunc func(*runtime.Scheme) error
 // by the time this function is called. If the CRD takes more than 5 seconds to
 // become ready, this function throws an error
 func AddToFrameworkScheme(addToScheme addToSchemeFunc, obj runtime.Object) error {
+	mutex.Lock()
+	defer func() { mutex.Unlock() }()
```

Comment:
Technically I think we could have only protected `addToScheme(Global.Scheme)` since the concurrent map write happens there. This will also block at most 10s on the poll for CRD creation.
But I think this okay for now. Probably safer too.